src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (i !== this.length - 1) {
+        string += this[i] + `${separator}`;
+      } else {
+        string += this[i];
+      }
+    }
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i++) {
      if (this[i] !== null || this[i] !== undefined) {
          if (i !== this.length - 1) {
          string += this[i] + `${separator}`;
        } else {
          string += this[i];
        }
      }
    }
```
Можна використати оператор ! при першому порівнянні, і тоді рядок this[i] = ''; буде не потрібен.

Reply:
![image](https://user-images.githubusercontent.com/88784433/134022169-c6af87e4-10fd-4346-90a4-020a7f1f6553.png)